docs/DeveloperGuide.md

```diff
@@ -441,43 +441,107 @@ considering code quality and end user experience.
 
 ### Update product feature
 
-The update product mechanism is facilitated by the `UpdateCommandParser`, `UpdateCommand`,  and the `UpdateProductDescriptor`.
-The `UpdateCommandParser` implements `Parser` to parse the user input, the `UpdateCommand` extends `Command` to execute the main logic,
-and the `UpdateProductDescriptor` allows the parser to pass a specification of the updated product to the `UpdateCommand`.
+#### What the update product feature does
+The update product feature allows users to modify the quantity or tags of existing products listed under suppliers and warehouses, or add new products to their existing lists.
+
+
+#### How it is implemented
+The update product mechanism is facilitated by 3 major components: `UpdateCommandParser`, `UpdateCommand`,  and the `UpdateProductDescriptor`.
```

Comment:
Minor issue but two blank space between the comma and last part of the sentence.